src/data_provider/CMakeLists.txt

```diff
@@ -159,8 +163,9 @@ elseif(APPLE)
   target_link_libraries(sysinfo cjson ${SRC_FOLDER}/external/libplist/bin/lib/libplist-2.0.a)
 endif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 
-target_link_libraries(sysinfo wazuhext)
+target_link_libraries(sysinfo wazuhext urlrequest)
 
+add_subdirectory(${SRC_FOLDER}/shared/http-request ${SRC_FOLDER}/shared/http-request/build)
```

Comment:
Shouldn't we use `add_subdirectory()` only for the folders we expect to contain a CMakeLists file?

Reply:
Sure, but I wonder why it doesn't fail here?